addons/account/models/account_full_reconcile.py

```diff
@@ -9,6 +9,16 @@ class AccountFullReconcile(models.Model):
     partial_reconcile_ids = fields.One2many('account.partial.reconcile', 'full_reconcile_id', string='Reconciliation Parts')
     reconciled_line_ids = fields.One2many('account.move.line', 'full_reconcile_id', string='Matched Journal Items')
     exchange_move_id = fields.Many2one('account.move', index="btree_not_null")
+    display_name = fields.Char(search='_search_display_name')
+
+    @api.depends('reconciled_line_ids.matching_number')
+    def _compute_display_name(self):
+        for rec in self:
+            rec.display_name = ', '.join(rec.reconciled_line_ids.filtered('matching_number').mapped('matching_number'))
```

Comment:
Won't they all share the same matching number anyway when linked to an account.full.reconcile ? So, you could do 

```suggestion
            rec.display_name = rec.reconciled_line_ids[0].matching_number
```